lxd/daemon.go

```diff
@@ -1424,7 +1424,7 @@ func (d *Daemon) init() error {
 	if metricsAddress != "" {
 		err = d.endpoints.UpMetrics(metricsAddress)
 		if err != nil {
-			return err
+			logger.Warn("Failed setting up metrics", logger.Ctx{"err": err})
```

Comment:
Line 1422 says:

> // Setup tertiary listeners that may use managed network addresses and must be started after networks.

Any ideas why this wasn't possible? Did you manage to recreate the issue?

As just logging a warning may just be glossing over an issue that original comment sought to avoid.

Reply:
> Any ideas why this wasn't possible? Did you manage to recreate the issue?

I was able to recreate this issue, but I don't know what caused it. However, this would have also failed if the address was already in use. So logging a warning should be fine here.

Reply:
Can you describe how you recreated the issue please?

Reply:
```
$ lxc network ls
+-----------------+----------+---------+----------------+---------------------------+-------------+---------+---------+
|      NAME       |   TYPE   | MANAGED |      IPV4      |           IPV6            | DESCRIPTION | USED BY |  STATE  |
+-----------------+----------+---------+----------------+---------------------------+-------------+---------+---------+
| br-int          | bridge   | NO      |                |                           |             | 0       |         |
+-----------------+----------+---------+----------------+---------------------------+-------------+---------+---------+
| enp0s31f6       | physical | NO      |                |                           |             | 0       |         |
+-----------------+----------+---------+----------------+---------------------------+-------------+---------+---------+
| enxe04f439357bf | physical | NO      |                |                           |             | 0       |         |
+-----------------+----------+---------+----------------+---------------------------+-------------+---------+---------+
| lxcbr0          | bridge   | NO      |                |                           |             | 0       |         |
+-----------------+----------+---------+----------------+---------------------------+-------------+---------+---------+
| lxdbr0          | bridge   | NO      |                |                           |             | 0       |         |
+-----------------+----------+---------+----------------+---------------------------+-------------+---------+---------+
| lxdbr1          | bridge   | YES     | 10.31.134.1/24 | fd42:581d:ede4:6b5e::1/64 |             | 4       | CREATED |
+-----------------+----------+---------+----------------+---------------------------+-------------+---------+---------+
| mpqemubr0       | bridge   | NO      |                |                           |             | 0       |         |
+-----------------+----------+---------+----------------+---------------------------+-------------+---------+---------+
| wlan0           | physical | NO      |                |                           |             | 0       |         |
+-----------------+----------+---------+----------------+---------------------------+-------------+---------+---------+

$ lxc config set core.metrics_address 10.31.134.2:8444        
Error: Cannot listen on http socket: Bind network address: listen tcp 10.31.134.2:8444: bind: cannot assign requested address
```

Despite the error, the config is saved. That's not a problem as such. But in this case, when LXD restarts, it cannot use `10.31.134.2` and will therefore not start the daemon but return an error.

Reply:
OK so that makes sense, you're setting it to an IP not on the lxdbr1 interface, so that should fail to bind, and we should not save the config - please can you fix that part too.

But if the IPs succeeds to bind when initially set (i.e if you set the bind address to 10.31.134.1) do you see LXD fail to start if you do a `lxd shutdown` such that lxdbr1 is removed? As that seems to be more what the OP was suggesting (although I've gone back to them for confirmation).

Reply:
> But if the IPs succeeds to bind when initially set (i.e if you set the bind address to 10.31.134.1) do you see LXD fail to start if you do a lxd shutdown such that lxdbr1 is removed?

LXD currently fails on startup if bind fails, even if it could be set previously. That's the reason we should change that to a warning.